docs/documentation.md

```diff
@@ -12,6 +12,53 @@ This connector is published as a Docker Image. The image name is `ghcr.io/hasura
 | HASURA_PLUGIN_LOG_LEVEL    | The log level. Possible values: `trace`, `debug`, `info`, `warn`, `error`, `fatal`, `panic`. Defaults to `info`                                                                                                      | false    | info                                                                                                  |
 | NDC_OAS_LAMBDA_PRETTY_LOGS | A Boolean flag to print human readable logs instead of JSON. Defaults to `false`                                                                                                                                     | false    | true                                                                                                  |
 
+## Adding Custom Headers to APIs
+
+You can use either of the following ways to add additional headers to the OpenAPI Connector which will be forwarded to your APIs
+
+### Via HML Files
+
+Since the OpenAPI Connector is simply a codegen mechnism on top of the NodeJS Connector, header forwarding works similarly for both. The OpenAPI Connector defines an optional `headers` variable of type `hasuraSdk.JSONValue` for every function in the `functions.ts` file. This variable, when configured in HML files, can recieve headers that are forwarded to it from the DDN Engine. [Read more about HTTP Header Forwarding in DDN](https://hasura.io/docs/3.0/recipes/business-logic/http-header-forwarding/).
+
+#### Example
+Add header forwarding to your OpenAPI Connector's `DataConnectorLink`
+```yaml
+kind: DataConnectorLink
+version: v1
+definition:
+  name: myopenapi
+  ...
+  headers:
+    Authorization:
+      valueFromEnv: APP_MYOPENAPI_AUTHORIZATION_HEADER # this header is received and consumed by the OpenAPI Connector. It is not forwarded to functions / API calls
+  
+  # setup header forwarding
+  argumentPresets:
+    - argument: headers # the variable in functions that will receive headers
+      value:
+        httpHeaders:
```

Comment:
Please aid my failing memory: we already have code in place to handle or flatten out the nested objects that would obtain here, right?

Reply:
The reason why we'd get nested objects in headers from the engine is because of a flawed value being set to that header. For example, we'd get an object for the following `argumentPreset`:
```yaml
  argumentPresets:
    - argument: headers
      value:
        httpHeaders:
          forward:
            - my-header
          additional: 
            auth: 
              literal:
                valueFromEnv: APP_MYOPENAPI_AUTHORIZATION_HEADER
``` 
The value for `auth` in this would be `{ "valueFromEnv": "APP_MYOPENAPI_AUTHORIZATION_HEADER" }`, because `literal` accepts a string value only. The engine does not currently support `valueFromEnv` for headers. 

The correct usage for `argumentPresets` would be:
```yaml
  argumentPresets:
    - argument: headers
      value:
        httpHeaders:
          forward:
            - my-header
          additional: 
            auth: 
              literal: my-breaer-token
```
Because of this, we don't need to flatten out headers that we get from the engine.